src/openrct2/world/footpath.c

```diff
@@ -449,7 +449,7 @@ money32 footpath_remove_real(sint32 x, sint32 y, sint32 z, sint32 flags)
 		sub_6A759F();
 	}
 
-	return (flags & (1 << 5)) || (gParkFlags & PARK_FLAGS_NO_MONEY) ? 0 : -MONEY(10,00);
+	return (flags & (1 << 5)) || (gParkFlags & PARK_FLAGS_NO_MONEY) || map_element_is_ghost(mapElement) ? 0 : -MONEY(10,00);
```

Comment:
```c
money32 cost = -MONEY(10,00);
bool is??? = (flags & (1 << 5);
bool moneyEnabled = (gParkFlags & PARK_FLAGS_NO_MONEY);
bool isGhost = map_element_is_ghost(mapElement);
if (is??? || moneyEnabled || isGhost) {
    cost = 0;
}
return cost;
```

The above approach ensures single entry, single exit principle and debugging is easier as each sub condition can be easily queried.